pkg/cmd/tknpac/generate/templates/java-template.yaml

```diff
@@ -0,0 +1,110 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: java-template
+  annotations:
+    # The event we are targeting as seen from the webhook payload
+    # this can be an array too, i.e: [pull_request, push]
+    pipelinesascode.tekton.dev/on-event: "pull_request"
+
+    # The branch or tag we are targeting (ie: main, refs/tags/*)
+    pipelinesascode.tekton.dev/on-target-branch: "main"
+
+    # Fetch the git-clone task from hub, we are able to reference later on it
+    # with taskRef and it will automatically be embedded into our pipeline.
+    pipelinesascode.tekton.dev/task: "git-clone"
+
+    # You can add more tasks in here to reuse, browse the one you like from here
+    # https://hub.tekton.dev/
+    # example:
+    # pipelinesascode.tekton.dev/task-2: "[maven, buildah]"
+
+    # How many runs we want to keep attached to this event
+    pipelinesascode.tekton.dev/max-keep-runs: "5"
+spec:
+  params:
+    # The variable with brackets are special to Pipelines as Code
+    # They will automatically be expanded with the events from Github.
+    - name: repo_url
+      value: "{{ repo_url }}"
+    - name: revision
+      value: "{{ revision }}"
+    - name: image_name
+      value: ""
+    - name: app_name
+      value: ""
+    - name: path_context
+      value: ""
+    - name: version
+      value: ""
+  pipelineSpec:
+    params:
+      - name: repo_url
+      - name: revision
+      - name: image_name
+      - name: app_name
+      - name: path_context
+      - name: version
+    workspaces:
+      - name: source
+      - name: basic-auth
+    tasks:
+      - name: fetch-repository
+        taskRef:
+          name: git-clone
+        workspaces:
+          - name: output
+            workspace: source
+          - name: basic-auth
+            workspace: basic-auth
+        params:
+          - name: url
+            value: $(params.repo_url)
+          - name: revision
+            value: $(params.revision)
+      - name: build
+        params:
+          - name: IMAGE
+            value: $(params.image_name)
+          - name: TLSVERIFY
+            value: "false"
+          - name: PATH_CONTEXT
+            value: $(params.path_context)
+          - name: VERSION
+            value: $(params.version)
+        runAfter:
+          - fetch-repository
+        taskRef:
+          kind: ClusterTask
+          name: s2i-java
+        workspaces:
+          - name: source
+            workspace: source
+      - name: deploy
+        params:
+          - name: SCRIPT
+            value: oc rollout status deploy/$(params.app_name)
```

Comment:
i missed that, those are actually deployment templates, on pac we want to have code oriented templates with for example a linter or unittesting, i think we can just do a `mvn test` instead in a template... cc @siamaksade if you have better idea